### DISCUSSION TITLE
Window startup location

### DISCUSSION THREAD

**Discussion by ollih1980** [AvaloniaUI/Avalonia#7104] [Q&A]
Hello Community,

I want to set the location of a new Window to my current mouse position.
In WPF it's possible via this.Left or this.Top.
Is there a way in Avalonia?

this.Position.X or this.Position.Y is "read-only".

Thanks a lot!

**Comment by MarkusKgit** [comment]
You can't set the X or Y seperately but you can set the Position as a whole. Take a look at this snippet:

```csharp
private void MainWindow_PointerPressed(object sender, PointerPressedEventArgs e)
{
    PixelPoint pixelPos = this.PointToScreen(e.GetPosition(this));
    Window wnd = new() { Title = "New Window", Width = 200, Height = 100, Position = pixelPos };
    wnd.ShowDialog(this);
}
```
![NewWindow](https://user-images.githubusercontent.com/13160892/145199945-d2ffc1ce-7059-4eb1-995a-711622473ef3.gif)

**Reply by ollih1980** [reply]
works like a charm! Thanks a lot!
